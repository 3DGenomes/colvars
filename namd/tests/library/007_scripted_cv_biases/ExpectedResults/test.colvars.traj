# step         d                     fa_d                   o                                                                                                 
           0    1.42467449615693e+01  1.87662751921535e+03  (  9.70743319377961e-01 ,  2.31771276816701e-01 ,  4.43248825020554e-02 , -4.44385859022720e-02 ) 
           2    1.45968734956943e+01  1.70156325215286e+03  (  9.64289355395474e-01 ,  2.48937082869898e-01 ,  8.30487263790259e-02 , -3.57669804432658e-02 ) 
           4    1.50568179643249e+01  1.47159101783754e+03  (  9.55472471437467e-01 ,  2.66275756821940e-01 ,  1.24167581770219e-01 , -2.74224231499856e-02 ) 
           6    1.56006447057904e+01  1.19967764710481e+03  (  9.44392616897005e-01 ,  2.83291169269234e-01 ,  1.65777259230496e-01 , -1.96621180578598e-02 ) 
           8    1.61967121734654e+01  9.01643913267277e+02  (  9.31616745008154e-01 ,  2.99186266886462e-01 ,  2.05971726695750e-01 , -1.23881366204203e-02 ) 
          10    1.68110806961106e+01  5.94459651944692e+02  (  9.18214277132689e-01 ,  3.12898195048884e-01 ,  2.42786784469456e-01 , -5.64252530568198e-03 ) 
          12    1.74094139082443e+01  2.95293045877836e+02  (  9.05544582913684e-01 ,  3.23726639699672e-01 ,  2.74208025347896e-01 , -1.73029617277426e-04 ) 
          14    1.79581926769606e+01  2.09036615196929e+01  (  8.94169246766526e-01 ,  3.31632480679141e-01 ,  3.00767078728152e-01 ,  4.51887695301882e-03 ) 
          16    1.84266767210664e+01 -2.13338360533184e+02  (  8.83965496033387e-01 ,  3.37521026567269e-01 ,  3.23435093570859e-01 ,  8.61966903279012e-03 ) 
          18    1.87897815291334e+01 -3.94890764566723e+02  (  8.74326627207205e-01 ,  3.42464244459866e-01 ,  3.43669760104206e-01 ,  1.27391606051312e-02 ) 
          20    1.90310815072806e+01 -5.15540753640325e+02  (  8.64853435016903e-01 ,  3.47470262417761e-01 ,  3.61959730859994e-01 ,  1.66765077446598e-02 ) 
# step         d                      ds                    fa_ds                  o                                                                                                  os                                                                                               
          20    1.42467449615693e+01   1.42467449615693e+01  1.87662751921535e+03  (  9.70743319377961e-01 ,  2.31771276816701e-01 ,  4.43248825020554e-02 , -4.44385859022720e-02 )  (  9.70743319377961e-01 ,  2.31771276816701e-01 ,  4.43248825020554e-02 , -4.44385859022720e-02 )
          22    1.45968769462860e+01   1.45968769462860e+01  1.70156152685700e+03  (  9.64277764519732e-01 ,  2.48913305181092e-01 ,  8.32458472821373e-02 , -3.57867051700994e-02 )  (  9.64277764519732e-01 ,  2.48913305181092e-01 ,  8.32458472821373e-02 , -3.57867051700994e-02 )
          24    1.50547997130750e+01   1.50547997130750e+01  1.47260014346250e+03  (  9.55568341695283e-01 ,  2.66049495288947e-01 ,  1.24002936606128e-01 , -2.70200318149417e-02 )  (  9.55568341695283e-01 ,  2.66049495288947e-01 ,  1.24002936606128e-01 , -2.70200318149417e-02 )
          26    1.55956320541925e+01   1.55956320541925e+01  1.20218397290375e+03  (  9.44596822093935e-01 ,  2.82769647554322e-01 ,  1.65650732983998e-01 , -1.83849061435584e-02 )  (  9.44596822093935e-01 ,  2.82769647554322e-01 ,  1.65650732983998e-01 , -1.83849061435585e-02 )
          28    1.61901797562312e+01   1.61901797562312e+01  9.04910121884400e+02  (  9.31755967347328e-01 ,  2.98566008314162e-01 ,  2.06346206471751e-01 , -1.05070960170485e-02 )  (  9.31755967347328e-01 ,  2.98566008314162e-01 ,  2.06346206471751e-01 , -1.05070960170485e-02 )
          30    1.68041700530110e+01   1.68041700530110e+01  5.97914973494500e+02  (  9.18427943555667e-01 ,  3.11952999656025e-01 ,  2.43229811253391e-01 , -3.83372134577825e-03 )  (  9.18427943555667e-01 ,  3.11952999656025e-01 ,  2.43229811253391e-01 , -3.83372134577825e-03 )
          32    1.74042349649180e+01   1.74042349649180e+01  2.97882517541000e+02  (  9.05786475242242e-01 ,  3.22570406141164e-01 ,  2.74767006836269e-01 ,  1.51205304056492e-03 )  (  9.05786475242242e-01 ,  3.22570406141164e-01 ,  2.74767006836269e-01 ,  1.51205304056492e-03 )
          34    1.79547674133106e+01   1.79547674133106e+01  2.26162933447007e+01  (  8.94550200769825e-01 ,  3.30184334490117e-01 ,  3.01196554538852e-01 ,  6.23531024958456e-03 )  (  8.94550200769825e-01 ,  3.30184334490117e-01 ,  3.01196554538852e-01 ,  6.23531024958456e-03 )
          36    1.84251934491198e+01   1.84251934491198e+01 -2.12596724559900e+02  (  8.84678288061678e-01 ,  3.35267779629636e-01 ,  3.23768005442535e-01 ,  1.06827536821974e-02 )  (  8.84678288061678e-01 ,  3.35267779629636e-01 ,  3.23768005442535e-01 ,  1.06827536821974e-02 )
          38    1.87900872289495e+01   1.87900872289495e+01 -3.95043614474750e+02  (  8.75354175193142e-01 ,  3.38944891381845e-01 ,  3.44467087494564e-01 ,  1.46237550272017e-02 )  (  8.75354175193142e-01 ,  3.38944891381845e-01 ,  3.44467087494564e-01 ,  1.46237550272017e-02 )
          40    1.90332078498876e+01   1.90332078498876e+01 -5.16603924943800e+02  (  8.66076104634896e-01 ,  3.42689420058431e-01 ,  3.63522144591631e-01 ,  1.81050476928932e-02 )  (  8.66076104634896e-01 ,  3.42689420058431e-01 ,  3.63522144591631e-01 ,  1.81050476928932e-02 )
