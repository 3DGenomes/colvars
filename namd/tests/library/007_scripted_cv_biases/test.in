Colvarstrajfrequency    2
Colvarsrestartfrequency 20000

colvar {
  name d

  lowerBoundary 12
  upperBoundary 32
  width 0.2

  outputAppliedForce

  distance {
    group1 { atomnumbers 4 }
    group2 { atomnumbers 99 }
  }
}

