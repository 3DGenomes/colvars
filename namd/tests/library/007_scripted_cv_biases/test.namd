# -*- tcl -*-

# FILENAMES
set outName             [file rootname [file tail [info script]]]
#set inFileNum           [expr [scan [string range $outName end-1 end] "%d"] - 1]
#set inName              [format "%s%02u" [string range $outName 0 end-2] $inFileNum]

# FILES

#outputname              test01
#DCDfile                 test01.dcd

# MD SETUP
timestep          0.5
numSteps		  20

# Interactive molecular dynamics
#IMDon               on
#IMDport             3000
#IMDfreq             20
#IMDwait             on

colvars			  on
colvarsConfig	  test.in
#colvarsInput  $inName.colvars.state
#colvarsInput  test.colvars.state

#tclForces on
#tclForcesScript {
#  addatom 12
#  proc calcforces {} {
#    addforce 12 { 10 0 0 }
#  }
#}

# INPUT
structure               ../Common/da.psf 
#parameters              missing.par
parameters              ../Common/par_all22_prot.inp
paraTypeCharmm          on
coordinates             ../Common/da.min.pdb

temperature 250
#bincoordinates       $inName.coor
#binvelocities        $inName.vel

# OUTPUT

outputname              ${outName}
outputenergies          100
outputtiming            2000
binaryoutput            no
# binaryrestart           yes

#XSTFreq                 1000

# DCD TRAJECTORY

DCDfile                 ${outName}.dcd
DCDfreq                 2000

seed		87654321

# CONSTANT-T
langevin                on
langevinTemp		    300.0
langevinDamping         10.0
langevinHydrogen	    on

# SPACE PARTITIONING
splitpatch              hydrogen
hgroupcutoff            2.8
stepspercycle           20
margin                  1.0

# CUT-OFFS
switching               on
switchdist              9.0
cutoff                  11.0
pairlistdist            12.0

# RESPA 
fullElectFrequency      1
nonbondedFreq           1

# 1-4 NON-BONDED
exclude                 scaled1-4
1-4scaling              1.0

# COM
commotion               no

# SHAKE
rigidBonds              none

# Scripted harmonic bias on colvar d

proc calc_colvar_forces {} {
  set center 18.
  set k 500.

  set d [cv colvar d value]
  set f [expr {$k * ($center - $d)}]
  set r [cv colvar d addforce $f]
  print "Applying force $r to colvar d"

  set r [cv colvar o addforce "(1., 0., 1., 0.)"]
  print "Applying force $r to colvar o"

  return
}
