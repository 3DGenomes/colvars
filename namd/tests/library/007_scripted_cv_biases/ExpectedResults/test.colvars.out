colvars: ----------------------------------------------------------------------
colvars: Using NAMD interface, version 2014-10-26.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "unscripted.in":
colvars: # indexFile = "" [default]
colvars: # analysis = off [default]
colvars: # colvarsTrajFrequency = 2
colvars: # colvarsRestartFrequency = 20000
colvars: # colvarsTrajAppend = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = d
colvars:   Initializing a new "distance" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # oneSiteSystemForce = off [default]
colvars:       Initializing atom group "group1".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group1" defined, 1 atoms initialized: total mass = 12.011.
colvars:       Initializing atom group "group2".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group2" defined, 1 atoms initialized: total mass = 12.011.
colvars:   All components initialized.
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # lowerWallConstant = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # upperWallConstant = 0 [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputSystemForce = off [default]
colvars:   # outputAppliedForce = on
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = o
colvars:   Initializing a new "orientation" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:       Initializing atom group "atoms".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "atoms" defined, 104 atoms initialized: total mass = 727.821.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = ../Common/da.pdb
colvars:     # refPositionsCol = "" [default]
colvars:     Centering the reference coordinates: it is assumed that each atom is the closest periodic image to the center of geometry.
colvars:     # closestToQuaternion = ( 1 , 0 , 0 , 0 ) [default]
colvars:   All components initialized.
colvars:   # width = 1 [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputSystemForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { d }
colvars:   # outputEnergy = off [default]
colvars:   # forceConstant = 500
colvars:   # centers = { 18 }
colvars:   # targetCenters = { 18 } [default]
colvars:   # targetForceConstant = 0 [default]
colvars:   # outputCenters = off [default]
colvars: ----------------------------------------------------------------------
colvars:   User forces script will be run at each bias update.
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: The final output state file will be "test.colvars.state".
colvars: Writing to colvar trajectory file "test.colvars.traj".
colvars: Saving collective variables state to "test.colvars.state".
colvars: Resetting the Collective Variables Module.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "scripted.in":
colvars: # indexFile = "" [default]
colvars: # analysis = off [default]
colvars: # colvarsTrajFrequency = 2
colvars: # colvarsRestartFrequency = 20000
colvars: # colvarsTrajAppend = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = ds
colvars:   Initializing a new "distance" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # oneSiteSystemForce = off [default]
colvars:       Initializing atom group "group1".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group1" defined, 1 atoms initialized: total mass = 12.011.
colvars:       Initializing atom group "group2".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "group2" defined, 1 atoms initialized: total mass = 12.011.
colvars:   All components initialized.
colvars:   This colvar uses scripted function "Id".
colvars:   # scriptedFunctionType = "scalar" [default]
colvars:   Expecting colvar value of type scalar number
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # lowerWallConstant = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # upperWallConstant = 0 [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputSystemForce = off [default]
colvars:   # outputAppliedForce = on
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = vec
colvars:   Initializing a new "cartesian" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 2
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:       Initializing atom group "atoms".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "atoms" defined, 2 atoms initialized: total mass = 24.022.
colvars:   Initializing a new "orientation" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:       Initializing atom group "atoms".
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # disableForces = off [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # enableFitGradients = on [default]
colvars:       Atom group "atoms" defined, 104 atoms initialized: total mass = 727.821.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = ../Common/da.pdb
colvars:     # refPositionsCol = "" [default]
colvars:     Centering the reference coordinates: it is assumed that each atom is the closest periodic image to the center of geometry.
colvars:     # closestToQuaternion = ( 1 , 0 , 0 , 0 ) [default]
colvars:   All components initialized.
colvars:   This colvar uses scripted function "vector".
colvars:   # scriptedFunctionType = vector
colvars:   Expecting colvar value of type n-dimensional vector
colvars:   # scriptedFunctionVectorSize = 5
colvars:   # width = 1 [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputSystemForce = off [default]
colvars:   # outputAppliedForce = on
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars:   User forces script will be run at each bias update.
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 0 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Saving collective variables state to "test.colvars.state".
